dts/bindings/mipi-dsi/mipi-dsi-device.yaml

```diff
@@ -0,0 +1,27 @@
+# Copyright (c) 2020 Teslabs Engineering S.L.
+# SPDX-License-Identifier: Apache-2.0
+
+# Common fields for MIPI-DSI devices
+
+include: base.yaml
+
+on-bus: mipi-dsi
+
+properties:
+    reg:
+      required: true
+
+    label:
+      required: true
+
+    data-lanes:
+      type: int
+      required: true
```

Comment:
I don't know anything about DSI, but the Linux bindings seem to use an array for this property; why is this an int?

Reply:
@mbolivar-nordic , can you please point me to the Linux bindings file.

Reply:
I'm just grepping around in linux/Documentation/dts/bindings/display. Here are a few places where data-lanes looks like it might be being used as an array of lanes rather than a number of lanes:

```
$ rg data-lanes
msm/dsi-controller-main.yaml
101:              data-lanes:
117:              data-lanes:
199:                                   data-lanes = <0 1 2 3>;

tegra/nvidia,tegra20-host1x.txt
129:	 - data-lanes: an array of data lane from 1 to 8. Valid array
530:								data-lanes = <1 2>;

bridge/sii902x.txt
18:	- sil,i2s-data-lanes: Array of up to 4 integers with values of 0-3
63:		sil,i2s-data-lanes = < 0 1 2 >;

bridge/renesas,dsi-csi2-tx.yaml
62:              data-lanes:
67:              - data-lanes
112:                    data-lanes = <1 2>;

bridge/ti,sn65dsi86.yaml
97:              data-lanes:
147:              lane-polarities: [data-lanes]
270:              data-lanes = <2 1 3 0>;

bridge/lontium,lt8912b.yaml
43:              data-lanes: true
46:              - data-lanes
87:              data-lanes = <0 1 2 3>;

bridge/toshiba,tc358775.yaml
135:                    data-lanes = <0 1 2 3>;
201:                    data-lanes = <0 1 2 3>;

bridge/analogix,anx7625.yaml
109:              data-lanes: true
159:                        data-lanes = <0 1 2 3>;

bridge/ti,sn65dsi83.yaml
50:              data-lanes:
70:              data-lanes:
146:                        data-lanes = <1 2 3 4>;
```

I'm not sure if this is applicable to your binding, however, since I see this binding is about the device side, not the host side. Just double checking with you on this since as I said I don't really know anything about the DSI protocol.

Reply:
@gmarull Can you help comment on this feedback from @mbolivar

Reply:
the property describes the number of data lanes available on the bus (up to 4 in MIPI-DSI).

What Linux does:

```
  data-lanes:
    $ref: /schemas/types.yaml#/definitions/uint32-array
    minItems: 1
    maxItems: 8
    items:
      # Assume up to 9 physical lane indices
      maximum: 8
    description:
      An array of physical data lane indexes. Position of an entry determines
      the logical lane number, while the value of an entry indicates physical
      lane, e.g. for 2-lane MIPI CSI-2 bus we could have "data-lanes = <1 2>;",
      assuming the clock lane is on hardware lane 0. If the hardware does not
      support lane reordering, monotonically incremented values shall be used
      from 0 or 1 onwards, depending on whether or not there is also a clock
      lane. This property is valid for serial busses only (e.g. MIPI CSI-2).
```

I don't think we need that description now, but Linux probably has chosen it for reasons. Considering Zephyr video/display support is in its infancy stage, I think we should follow Linux.

Reply:
Sorry, I was not clear on your earlier comment. I have updated to convert this to a array.